src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -9182,6 +9182,8 @@ void Unit::Mount(uint32 mount, uint32 VehicleId, uint32 creatureEntry)
     }
 
     RemoveAurasWithInterruptFlags(AURA_INTERRUPT_FLAG_MOUNT);
+    if (GetTypeId() == TYPEID_UNIT)
```

Comment:
Don't use AI as an example, IsAIEnabled is a separate boolean to disable AI (yes, its a valid state) - also its never null for creatures